Formula for different versions

jmccann opened the pull request:
This is my attempt to fix #3 

This PR does not do exactly what is outlined in #3 as it does not seem possible to pass options to brew formula with values.  Options seem to be strictly true/false flags.

Reading up on how [versions are handled in brew](http://docs.brew.sh/Versions.html) and looking at what the php tap does to handle versions that have similar code with ["abstracting"]() that is the path I chose to take/copy.  :)

Examples of commands at work:

```bash
$ brew search drone
drone/drone/drone                        drone/drone/drone@0.5
$ brew install drone
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (caskroom/cask).
No changes to formulae.

==> Installing drone from drone/drone
==> Downloading http://downloads.drone.io/latest/release/darwin/amd64/drone.tar.

curl: (22) The requested URL returned error: 403 Forbidden
Error: Failed to download resource "drone"
Download failed: http://downloads.drone.io/latest/release/darwin/amd64/drone.tar.gz
$ brew install drone@0.5
==> Installing drone@0.5 from drone/drone
==> Downloading http://downloads.drone.io/0.5.0/release/darwin/amd64/drone.tar.g
Already downloaded: /Users/z087177/Library/Caches/Homebrew/drone@0.5-0.5.tar.gz
🍺  /usr/local/Cellar/drone@0.5/0.5: 3 files, 18.8MB, built in 0 seconds
```

kzaitsev commented on Abstract/abstract-drone.rb:
```diff
@@ -0,0 +1,35 @@
+require "formula"
+
+class AbstractDrone < Formula
+  def self.init
```
http://ruby-doc.org/core-2.0.0/Class.html#method-i-inherited

Instead manually initialization

jmccann replied:
Thanks for the suggestion!  Just tried that now but seems to not work.

```bash
$ brew install drone
Error: formulae require at least a URL
Please report this bug:
  http://docs.brew.sh/Troubleshooting.html
/usr/local/Homebrew/Library/Homebrew/formula.rb:236:in `determine_active_spec'
/usr/local/Homebrew/Library/Homebrew/formula.rb:190:in `initialize'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:88:in `new'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:88:in `get_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:263:in `factory'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:436:in `find_with_priority'
/usr/local/Homebrew/Library/Homebrew/extend/ARGV.rb:47:in `block in formulae'
/usr/local/Homebrew/Library/Homebrew/extend/ARGV.rb:43:in `map'
/usr/local/Homebrew/Library/Homebrew/extend/ARGV.rb:43:in `formulae'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:108:in `install'
/usr/local/Homebrew/Library/Homebrew/brew.rb:90:in `<main>'
```

aaronkorver commented:
Is there any update on this?  I'd love to bring in different versions with brew.

bradrydzewski commented:
@aaronkorver the binaries are now distributed as github releases. The brew formula would therefore need to be updated accordingly. https://github.com/drone/drone-cli/releases

jmccann commented:
I'll take a look at that!

jmccann commented:
I've updated the formula to look for latest at the github releases for `drone/drone-cli`.

I also added versions for `0.6` and `0.7`.

@bradrydzewski

jmccann commented:
@Bugagazavr wouldn't mind a quick review again from you since I made some updates
